### PR TITLE
Benface/Nextra

### PR DESCRIPTION
**Pull request by benface**
Some proposed changes to accompany my review of #301...

### CODE REVIEW

**Comment by saihaj** [comment]
@benface if you can rebase from `nextra` branch and just have your change commits cause right now the diff is noise cause of the changes from `main` which were not synced yet

**Comment by benface** [comment]
@saihaj – I had intentionally merged `main` into my branch, but I guess that's problematic, especially with squashing commits. I just did like you said. Better now?

**Comment by saihaj** [comment]
> @saihaj – I had intentionally merged `main` into my branch, but I guess that's problematic, especially with squashing commits. I just did like you said. Better now?

yeah now we can easily go through commits!

File: website/components/EditPageLink.tsx [hunk 1]
```diff
@@ -16,12 +17,12 @@ export const EditPageLink = ({ mobile = false, ...props }: EditPageLinkProps) =>
...
   const { pagePath: _pagePath } = useContext(NavContext)!
   const pagePathSegments = _pagePath.split('/')
-  pagePathSegments[0] = ['en', '[locale]'].includes(pagePathSegments[0]) ? pagePathSegments[0] : 'en'
+  pagePathSegments[1] = ['en', '[locale]'].includes(pagePathSegments[1]) ? pagePathSegments[1] : 'en'
   const pagePath = pagePathSegments.join('/')
 
   return (
     <Link
-      href={`https://github.com/graphprotocol/docs/blob/main/pages/${pagePath}`}
+      href={`https://github.com/graphprotocol/docs/blob/main/website/${pagePath}`}
```
Comment (B2o5T) [inline website/components/EditPageLink.tsx:25]: nice catch for edit on github, forget to test it

File: website/tsconfig.json [hunk 2]
```diff
@@ -1,11 +1,4 @@
 {
   "extends": "../tsconfig.json",
-  "compilerOptions": {
-    "baseUrl": ".",
-    "paths": {
-      "@/*": ["*"]
```
Comment (saihaj) [inline website/tsconfig.json:6]: I think this was nice :)

File: website/next.config.mjs [hunk 3]
```diff
@@ -20,13 +20,14 @@ const withNextra = nextra({
   },
   transform(content, { route }) {
     if (route) {
+      const routeSegments = route.split('/')
       return `
-import { getNavItems } from '@/navigation'
+import { getNavItems } from '${'../'.repeat(routeSegments.length - 1)}navigation'
```
Comment (B2o5T) [inline website/next.config.mjs:25]: why did you remove `@` alias in tsconfig? seems it worked well?

Comment (benface) [reply website/next.config.mjs:25]: It didn't! Try running `pnpm lint` in your branch.

**Review by dimaMachina — APPROVED** [review]

File: website/tsconfig.json (hunk 2 above)
Comment (benface) [reply website/tsconfig.json:6]: I agree, if you find a way to make it pass `pnpm lint`, let's put it back!

File: website/next.config.mjs (hunk 3 above)
Comment (saihaj) [reply website/next.config.mjs:25]: I think ESlint config has a way to pick aliases 🤔

**Review by saihaj — APPROVED** [review]

File: website/next.config.mjs (hunk 3 above)
Comment (B2o5T) [reply website/next.config.mjs:25]: image

Comment (benface) [reply website/next.config.mjs:25]: Oh yeah Prettier fails which prevents `tsc` from even running... try `pnpm lint:fix`

Comment (benface) [reply website/next.config.mjs:25]: We should probably replace the `&&`s with `;` so each command runs regardless of the result of the previous one.

Comment (B2o5T) [reply website/next.config.mjs:25]: to fix you can replace the below in `package.json`

```diff
-"typecheck": "tsc --noEmit",
+"typecheck": "pnpm --filter @graphprotocol/docs typecheck",
```

Comment (benface) [reply website/next.config.mjs]: Ohhh, thank you @B2o5T! Done in https://github.com/graphprotocol/docs/pull/319/commits/a0717fa035aa6a0ed4fc4524bf737a0e270db4d9, and then I reverted removing the alias in https://github.com/graphprotocol/docs/pull/319/commits/e8baaf649f0933da1b572dfba25ce2fe0aab29fe.

By the way, there are still 4 ESLint warnings:

CleanShot 2023-03-16 at 22 20 47@2x

Those are weird, I don't really understand why TypeScript thinks they are `any`.

Comment (B2o5T) [reply website/next.config.mjs]: I will take a look tomorrow!

File: package.json [hunk 4]
```diff
@@ -7,11 +7,11 @@
     "build": "pnpm -r build",
     "start": "pnpm --filter @graphprotocol/docs start",
     "export": "pnpm --filter @graphprotocol/docs export",
-    "lint": "eslint . --ext .js,.jsx,.ts,.tsx,.mjs && pnpm prettier:check && pnpm typecheck",
-    "lint:fix": "eslint . --fix --ext .js,.jsx,.ts,.tsx,.mjs && pnpm prettier && pnpm typecheck",
+    "lint": "eslint . --ext .js,.jsx,.ts,.tsx,.mjs; pnpm prettier:check; pnpm typecheck",
+    "lint:fix": "eslint . --fix --ext .js,.jsx,.ts,.tsx,.mjs; pnpm prettier; pnpm typecheck",
```
Comment (B2o5T) [inline package.json:11]: @benface we should use `&&` instead, otherwise, it's not fails if some command fails, e.g. `eslint --max-warnings 0`

Comment (benface) [reply package.json]: Ah yeah, oops. I guess I just find it annoying that if e.g. Prettier fails, the TS errors are not even shown. Maybe Prettier should be last?

Comment (B2o5T) [reply package.json]: yes, I guess the following commands `typecheck` -> `eslint` -> `prettier` will be better
